server/api/api.go

```diff
@@ -2039,6 +2040,129 @@ func (a *API) handleUploadFile(w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("fileID", fileID)
 	auditRec.Success()
 }
+func (a *API) handleTeamBoardsInsights(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation GET /teams/{teamID}/boards/insights getTeamUsers
+	//
+	// Returns team boards insights
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: teamID
+	//   in: path
+	//   description: Team ID
+	//   required: true
+	//   type: string
+	// - name: duration
+	//   in: query
+	//   description: duration of data to calculate insights for
+	//   required: true
+	//   type: string
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       type: array
+	//       items:
+	//         "$ref": "#/definitions/BoardInsight"
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	vars := mux.Vars(r)
+	teamID := vars["teamID"]
+	userID := getUserID(r)
+	query := r.URL.Query()
+	duration := query.Get("duration")
+
+	if !a.permissions.HasPermissionToTeam(userID, teamID, model.PermissionViewTeam) {
+		a.errorResponse(w, r.URL.Path, http.StatusForbidden, "Access denied to team", PermissionError{"access denied to team"})
+		return
+	}
+
+	auditRec := a.makeAuditRecord(r, "getTeamBoardsInsights", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelRead, auditRec)
+
+	boardsInsights, err := a.app.GetTeamBoardsInsights(teamID, duration)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "duration="+duration, err)
+		return
+	}
+
+	data, err := json.Marshal(boardsInsights)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	jsonBytesResponse(w, http.StatusOK, data)
+
+	auditRec.AddMeta("boardInsightCount", len(boardsInsights))
+	auditRec.Success()
+}
+
+func (a *API) handleUserBoardsInsights(w http.ResponseWriter, r *http.Request) {
```

Comment:
This looks leaky, there is no permission check to verify if you can see other users insights. I would expect at least to check the permissions for that (I don't know if a system_admin permission, or you can check your own insights).

In any case, I expect here some permission checks.